swarm/src/behaviour.rs

```diff
@@ -294,6 +294,9 @@ pub trait PollParameters {
     /// The iterator's elements are the ASCII names as reported on the wire.
     ///
     /// Note that the list is computed once at initialization and never refreshed.
+    #[deprecated(
+        note = "Use `libp2p_swarm::SupportedProtocols` in your `ConnectionHandler` instead."
+    )]
     fn supported_protocols(&self) -> Self::SupportedProtocolsIter;
```

Comment:
With this deprecation, the entire `PollParameters` trait is deprecated. Should we remove it at the same time?

It would be an easier migration if users can still use this but get a deprecation warning and only remove it in the breaking change after. What do you think @mxinden?

Reply:
I don't see a benefit in removing it right away, i.e. I don't think its existence is blocking any other changes. Thus, yes, deprecating it for now sounds good to me.